core/policy.go

```diff
@@ -54,7 +55,22 @@ func NewMessageQueuePolicy(messages MessageProvider, maxAge uint) *DefaultQueueP
 
 // HandleNewHead removes from the queue all messages that have now been mined in new blocks.
 func (p *DefaultQueuePolicy) HandleNewHead(ctx context.Context, target PolicyTarget, oldTips, newTips []types.TipSet) error {
-	// Rearrange the tipsets into increasing height order so messages are discovered in nonce order.
+	chainHeight := uint64(0)
+	var err error
+	if len(newTips) > 0 {
+		chainHeight, err = newTips[0].Height()
+		if err != nil {
+			return err
+		}
+	} else if len(oldTips) > 0 { // A pure rewind is unlikely in practice.
```

Comment:
This is a little weird. Seems like the only consequence of this clause is that the message requeued will be stamped with the maximum old chain height. It would be more accurate to use the minimum old chain height. Like you say, this is unlikely (impossible?) and the consequences seem small, so seems fine.

Reply:
I opted for the maximum as a "safe" option, to prevent premature expiration.